dbs/datasets.go

```diff
@@ -675,46 +717,84 @@ func (a *API) UpdateDatasets() error {
 			createBy = t[0]
 		}
 	}
+
+	date := time.Now().Unix()
+
 	var isValidDataset int64
 	var dataset string
 	var datasetAccessType string
-	if v, ok := a.Params["dataset"]; ok {
+	var physicsGroupName string
+	if v, ok := a.Params["dataset_access_type"]; ok {
+		tmpl["DatasetAccessType"] = true
 		switch t := v.(type) {
 		case string:
-			dataset = t
+			datasetAccessType = t
 		case []string:
-			dataset = t[0]
+			datasetAccessType = t[0]
+		}
+		if datasetAccessType == "" {
+			msg := "invalid datasetAccessType parameter"
+			return Error(InvalidParamErr, ParametersErrorCode, msg, "dbs.datasets.UpdateDatasets")
+		}
+		if err := CheckPattern("dataset_access_type", datasetAccessType); err != nil {
+			msg := "datasetAccessType parameter pattern invalid"
+			return Error(err, PatternErrorCode, msg, "dbs.datasets.UpdateDatasets")
 		}
 	}
-	if v, ok := a.Params["dataset_access_type"]; ok {
+
+	if v, ok := a.Params["physics_group_name"]; ok {
+		tmpl["PhysicsGroup"] = true
 		switch t := v.(type) {
 		case string:
-			datasetAccessType = t
+			physicsGroupName = t
 		case []string:
-			datasetAccessType = t[0]
+			physicsGroupName = t[0]
+		}
+		if physicsGroupName == "" {
+			msg := "invalid physics_group parameter"
+			return Error(InvalidParamErr, ParametersErrorCode, msg, "dbs.datasets.UpdateDatasets")
+		}
+		if err := CheckPattern("physics_group", physicsGroupName); err != nil {
+			msg := "physics_group_name parameter pattern invalid"
+			return Error(err, PatternErrorCode, msg, "dbs.datasets.UpdateDatasets")
 		}
 	}
-	date := time.Now().Unix()
 
-	// validate input parameters
-	if dataset == "" {
-		msg := "invalid dataset parameter"
-		return Error(InvalidParamErr, ParametersErrorCode, msg, "dbs.datasets.UpdateDatasets")
+	if v, ok := a.Params["dataset"]; ok {
```

Comment:
Since it is already 3rd time I see such code structure in this codebase I suggest to put it into stand-along function and use it across the code, e.g.
```
func testParam(params Params, key string) {
   if v, ok := params[key]; ok {
		switch t := v.(type) {
		case string:
			dataset = t
		case []string:
			dataset = t[0]
		}
		if dataset == "" {
			msg := "invalid dataset parameter"
			return Error(InvalidParamErr, ParametersErrorCode, msg, "dbs.datasets.UpdateDatasets")
		}
		if err := CheckPattern("dataset", dataset); err != nil {
			msg := "dataset parameter pattern invalid"
			return Error(err, PatternErrorCode, msg, "dbs.datasets.UpdateDatasets")
		}
  }
}
```
Then, you will call it like:
```
testParam(a.Params, "dataset_access_type')
testParam(a.Params, "physics_group')
testParam(a.Params, "dataset')

```